Index: handlers/solr.php
===================================================================
--- handlers/solr.php	(revision 1038)
+++ handlers/solr.php	(working copy)
@@ -74,9 +74,11 @@
      */
     protected function connect()
     {
-        $this->connection = @stream_socket_client( "tcp://{$this->host}:{$this->port}" );
+    	$errno = $errstr = false;
+        $this->connection = @stream_socket_client( "tcp://{$this->host}:{$this->port}", $errno, $errstr, 5 );
         if ( !$this->connection )
         {
+        	return false;
             throw new ezcSearchCanNotConnectException( 'solr', "http://{$this->host}:{$this->port}{$this->location}" );
         }
     }
