Index: acp_mods.php
===================================================================
--- acp_mods.php	(revision 909)
+++ acp_mods.php	(working copy)
@@ -1349,6 +1349,7 @@
 				{
 					$template->assign_block_vars('edit_files', array(
 						'FILENAME'	=> $filename,
+						'S_SUCCESS'	=> false,
 					));
 
 					$status = $editor->open_file($filename, $this->backup_root);
@@ -1362,6 +1363,8 @@
 						continue;
 					}
 
+					$edit_success = true;
+
 					foreach ($edits as $finds)
 					{
 						$comment = '';
@@ -1378,10 +1381,10 @@
 								continue;
 							}
 
-							$template->assign_block_vars('edit_files.finds', array(
+							$find_tpl = array(
 								'FIND_STRING'	=> htmlspecialchars($find),
 								'COMMENT'		=> htmlspecialchars($comment),
-							));
+							);
 
 							$offset_ary = $editor->find($find);
 
@@ -1529,8 +1532,13 @@
 									break;
 								}
 
+								$template->assign_block_vars('edit_files.finds', array_merge($find_tpl, array(
+									'S_SUCCESS'		=> $status,
+								)));
+
 								if (!$status)
 								{
+									$edit_success = false;
 									$mod_installed = false;
 								}
 
@@ -1561,6 +1569,10 @@
 
 						$editor->close_edit();
 					}
+
+					$template->alter_block_array('edit_files', array(
+						'S_SUCCESS'		=> $edit_success,
+					), true, 'change');
 				}
 
 				if ($change)
