Index: acp_mods.php
===================================================================
--- acp_mods.php	(revision 924)
+++ acp_mods.php	(working copy)
@@ -550,19 +550,6 @@
 					$processed_templates = array('prosilver');
 					$processed_templates += explode(',', $row['mod_template']);
 
-					// now grab the templates that have not already been processed
-					$sql = 'SELECT template_id, template_path FROM ' . STYLES_TEMPLATE_TABLE . '
-						WHERE ' . $db->sql_in_set('template_name', $processed_templates, true);
-					$result = $db->sql_query($sql);
-
-					while ($row = $db->sql_fetchrow($result))
-					{
-						$template->assign_block_vars('board_templates', array(
-							'TEMPLATE_ID'		=> $row['template_id'],
-							'TEMPLATE_NAME'		=> $row['template_path'],
-						));
-					}
-
 					$s_hidden_fields = build_hidden_fields(array(
 						'action'	=> 'install',
 						'parent'	=> $parent_id,
@@ -1908,32 +1895,7 @@
 
 	function handle_template_prompt(&$children, &$elements, $action)
 	{
-		global $db, $template, $phpbb_root_path, $parent_id;
-
-		if (isset($children['template']) && sizeof($children['template']))
-		{
-			// additional styles are available for this MOD
-			$sql = 'SELECT template_id, template_name FROM ' . STYLES_TEMPLATE_TABLE;
-			$result = $db->sql_query($sql);
-
-			$installed_templates = array();
-			while ($row = $db->sql_fetchrow($result))
-			{
-				$installed_templates[$row['template_id']] = $row['template_name'];
-			}
-			$db->sql_freeresult($result);
-
-			foreach ($children['template'] as $key => $tag)
-			{
-				// remove useless title from MODX 1.2.0 tags
-				$children['template'][$tag['realname']] = is_array($tag) ? $tag['href'] : $tag;
-			}
-
-			$available_templates = array_keys($children['template']);
-
-			// $process_templates are those that are installed on the board and provided for by the MOD
-			$process_templates = $elements['template'] = array_intersect($available_templates, $installed_templates);
-		}
+		return;
 	}
 
 	function upload_mod()
Index: editor.php
===================================================================
--- editor.php	(revision 924)
+++ editor.php	(working copy)
@@ -107,6 +107,11 @@
 	{
 		global $phpbb_root_path, $db, $user;
 
+		if (strpos($filename, '..') !== false)
+		{
+			return $user->lang['FILE_EMPTY'];
+		}
+
 		$this->file_contents = @file($phpbb_root_path . $filename);
 
 		if ($this->file_contents === false)
@@ -116,41 +121,7 @@
 
 		$this->file_contents = $this->normalize($this->file_contents);
 
-		// Check for file contents in the database if this is a template file
-		// this will overwrite the @file call if it exists in the DB. 
-		if (strpos($filename, 'template/') !== false)
-		{
-			// grab template name and filename
-			preg_match('#styles/([a-z0-9_]+)/template/([a-z0-9_]+.[a-z]+)#i', $filename, $match);
-
-			$sql = 'SELECT d.template_data, d.template_id 
-				FROM ' . STYLES_TEMPLATE_DATA_TABLE . ' d, ' . STYLES_TEMPLATE_TABLE . " t
-				WHERE d.template_filename = '" . $db->sql_escape($match[2]) . "'
-					AND t.template_id = d.template_id
-					AND t.template_storedb = 1
-					AND t.template_name = '" . $db->sql_escape($match[1]) . "'";
-			$result = $db->sql_query($sql);
-
-			if ($row = $db->sql_fetchrow($result))
-			{
-				$this->file_contents = explode("\n", $this->normalize($row['template_data']));
-
-				// emulate the behavior of file()
-				$lines = sizeof($this->file_contents);
-				for ($i = 0; $i < $lines; $i++)
-				{
-					$this->file_contents[$i] .= "\n";
-				}
-
-				$this->template_id = $row['template_id'];
-			}
-			else
-			{
-				$this->template_id = 0;
-			}
-		}
-
-		/* 
+		/*
 		* If the file does not exist, or is empty, die.
 		* Non existant files cannot be edited, and empty files will have no
 		* finds
Index: functions_mods.php
===================================================================
--- functions_mods.php	(revision 924)
+++ functions_mods.php	(working copy)
@@ -206,24 +206,7 @@
 */
 function update_database_template($filename, $template_id, $file_contents, $install_time)
 {
-	global $db;
-
-	// grab filename
-	preg_match('#styles/[a-z0-9_]+/template/([a-z0-9_]+.html)#i', $filename, $match);
-
-	if (empty($match[1]))
-	{
-		return false;
-	}
-
-	$sql = 'UPDATE ' . STYLES_TEMPLATE_DATA_TABLE . "
-		SET template_data = '" . $db->sql_escape($file_contents) . "', template_mtime = " . (int) $install_time . '
-		WHERE template_id = ' . (int) $template_id . "
-		AND template_filename = '" . $db->sql_escape($match[1]) . "'";
-	$db->sql_query($sql);
-
-	// if something failed, sql_query will error out
-	return true;
+	return;
 }
 
 function determine_write_method($pre_install = false)
Index: mod_parser.php
===================================================================
--- mod_parser.php	(revision 941)
+++ mod_parser.php	(working copy)
@@ -99,43 +99,6 @@
 	*/
 	function parse_sql(&$sql_query)
 	{
-		global $dbms, $table_prefix;
-
-		if (!function_exists('get_available_dbms'))
-		{
-			global $phpbb_root_path, $phpEx;
-
-			include($phpbb_root_path . 'includes/functions_install.' . $phpEx);
-		}
-
-		static $available_dbms;
-
-		if (!isset($available_dbms))
-		{
-			$available_dbms = get_available_dbms($dbms);
-		}
-
-		$remove_remarks = $available_dbms[$dbms]['COMMENTS'];
-		$delimiter = $available_dbms[$dbms]['DELIM'];
-
-		if (sizeof($sql_query) == 1)
-		{
-			// do some splitting here
-			$sql_query = preg_replace('#phpbb_#i', $table_prefix, $sql_query);
-			$remove_remarks($sql_query[0]);
-			$sql_query = split_sql_file($sql_query[0], $delimiter);
-		}
-		else
-		{
-			$query_count = sizeof($sql_query);
-			for ($i = 0; $i < $query_count; $i++)
-			{
-				$sql_query[$i] = preg_replace('#phpbb_#i', $table_prefix, $sql_query[$i]);
-				$remove_remarks($sql_query[$i]);
-			}
-		}
-
-		//return $sql_query;
 	}
 
 	/**