--- AutoMOD/includes/acp/acp_mods.php (1.0.0.1)
+++ titania/includes/library/automod/acp_mods.php
@@ -1649,8 +1641,13 @@
 			}
 		} // end foreach
 
+		if (!$mod_installed)
+		{
+			$template->assign_var('S_DISPLAY_FILE_EDITS', true);
+		}
+
 		// Move included files
-		if (isset($actions['NEW_FILES']) && !empty($actions['NEW_FILES']) && $change && ($mod_installed || $force_install))
+		if (isset($actions['NEW_FILES']) && !empty($actions['NEW_FILES']) && ($mod_installed || $force_install))
 		{
 			$template->assign_var('S_NEW_FILES', true);
 
@@ -1661,6 +1658,7 @@
 				if ($status !== true && !is_null($status))
 				{
 					$mod_installed = false;
+					$template->assign_var('S_DISPLAY_NEW_FILES', true);
 				}
 
 				$template->assign_block_vars('new_files', array(

--- AutoMOD/includes/editor.php (1.0.0.1)
+++ titania/includes/library/automod/editor.php
@@ -770,7 +740,7 @@
 
 		if (strpos($from, $phpbb_root_path) !== 0)
 		{
-			$from = $phpbb_root_path . $from;
+			//$from = $phpbb_root_path . $from;
 		}
 
 		// When installing a MODX 1.2.0 MOD, this happens once in a long while.
@@ -799,6 +769,16 @@
 		if (empty($files))
 		{
 			return false;
+		}
+
+		// We're only interested in finding out whether the file/directory exists for titania's checks
+		if (!file_exists($from))
+		{
+			return false;
+		}
+		else
+		{
+			return true;
 		}
 
 		// Look at the last character of $to and compare it to '/'
