Index: zend_lucene.php
===================================================================
--- zend_lucene.php	(revision 593)
+++ zend_lucene.php	(working copy)
@@ -67,7 +67,7 @@
         // Set proper default encoding for query parser
         Zend_Search_Lucene_Search_QueryParser::setDefaultEncoding( 'UTF-8' );
         Zend_Search_Lucene_Analysis_Analyzer::setDefault( 
-            new Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8Num()
+            new Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8Num_CaseInsensitive()
         );
     }
 
